textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/common/IJoinPredicate.java

```diff
@@ -17,4 +18,6 @@
 
 	ITuple joinTuples(ITuple outerTuple, ITuple innerTuple, Schema outputSchema)
 			throws Exception;
+	
+	Schema generateOutputSchema(Schema innerOperatorSchema, Schema outerOperatorSchema) throws DataFlowException;
```

Comment:
`generateOutputSchema` function is moved from `Join` to the predicates, since different join predicates may produce different output schema